include/spdlog/details/async_log_helper.h

```diff
@@ -249,7 +254,12 @@ inline bool spdlog::details::async_log_helper::process_next_msg(clock::time_poin
     }
     else //empty queue
     {
-        sleep_or_yield(last_pop);
+        auto time_since_op = sleep_or_yield(last_pop);
+        if (_auto_flush_millis > std::chrono::milliseconds::zero() && time_since_op > _auto_flush_millis)
+        {
+            for (auto &s : _sinks)
+                s->flush();
```

Comment:
There is a problem here. flush() will be called continuously as long as the queue is empty, not just every _auto_flush_millis.